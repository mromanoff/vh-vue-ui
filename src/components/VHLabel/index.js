import VHLabel from './VHLabel.vue';

export default VHLabel;
