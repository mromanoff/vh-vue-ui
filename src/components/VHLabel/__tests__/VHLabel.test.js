import { shallowMount } from '@vue/test-utils';
import VHLabel from '../VHLabel.vue';

describe('VHLabel.vue', () => {
  let wrapper;

  // TEST functional component with passing props: https://vue-test-utils.vuejs.org/api/options.html#context

  beforeEach(() => {
    wrapper = shallowMount(VHLabel, {
      context: {
        props: { size: 'medium', color: 'grey' },
      },
      slots: {
        default: 'Test VHLabel',
      },
    });
  });

  it('renders default VHLabel', () => {
    expect(wrapper.isFunctionalComponent).toBe(true);
    expect(wrapper.is('div')).toBe(true);
    expect(wrapper.classes()).toStrictEqual(['VHLabel', 'VHLabel--grey']);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Green VHLabel', () => {
    wrapper = shallowMount(VHLabel, {
      context: {
        props: { color: 'green' },
      },
      slots: {
        default: 'Test Green VHLabel',
      },
    });
    expect(wrapper.classes()).toStrictEqual(['VHLabel', 'VHLabel--green']);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Blue VHLabel', () => {
    wrapper = shallowMount(VHLabel, {
      context: {
        props: { color: 'blue' },
      },
      slots: {
        default: 'Test Blue VHLabel',
      },
    });
    expect(wrapper.classes()).toStrictEqual(['VHLabel', 'VHLabel--blue']);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Black VHLabel', () => {
    wrapper = shallowMount(VHLabel, {
      context: {
        props: { color: 'black' },
      },
      slots: {
        default: 'Test Black VHLabel',
      },
    });
    expect(wrapper.classes()).toStrictEqual(['VHLabel', 'VHLabel--black']);
    expect(wrapper).toMatchSnapshot();
  });
});
