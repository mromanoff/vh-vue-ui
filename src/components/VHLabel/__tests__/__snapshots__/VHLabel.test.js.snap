// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VHLabel.vue renders Black VHLabel 1`] = `<div class="VHLabel VHLabel--black">Test Black VHLabel</div>`;

exports[`VHLabel.vue renders Blue VHLabel 1`] = `<div class="VHLabel VHLabel--blue">Test Blue VHLabel</div>`;

exports[`VHLabel.vue renders Green VHLabel 1`] = `<div class="VHLabel VHLabel--green">Test Green VHLabel</div>`;

exports[`VHLabel.vue renders default VHLabel 1`] = `<div class="VHLabel VHLabel--grey">Test VHLabel</div>`;
